From 263ff38d5ef4daf0d88b7deaffdf669211af1481 Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Tue, 5 Mar 2019 09:08:43 +0100
Subject: [PATCH 2/2] [DNM] Squash of lineage-16.0-android-9.0.0_r34

Squashed commit of the following:

commit 68f7efc044b4ff9813fba79e3e16e5804c4df651
Merge: b32be3f07 973b64c3e
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Thu Jan 17 21:08:24 2019 +0000

    Merge cherrypicks of [6086654] into pi-qpr2-release

    Change-Id: I978a6db494532b84be98221b8cf413ca266cab59

commit 973b64c3e55be7f8e14cece946b0c726a31ef2b4
Author: Tim Murray <timmurray@google.com>
Date:   Tue Jan 8 11:01:18 2019 -0800

    Remove some PR_SET_VMAs during pthread_create

    PR_SET_VMA takes mmap_sem, which can cause contention and reduce
    performance any time many threads are created at the same time,
    like app startup.

    Test: camera launch performance
    Bug: 122471935
    Change-Id: If7fa7ad99654c01d503f694976fd92bfd30d2afd
    Merged-in: If7fa7ad99654c01d503f694976fd92bfd30d2afd
    (cherry picked from commit dad73ef232034dd0f4f46bda90c74e8727d840cc)

Change-Id: I56bb2f4fc44eb734a434f84e6aac3286d3640e58
---
 libc/bionic/pthread_create.cpp | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/libc/bionic/pthread_create.cpp b/libc/bionic/pthread_create.cpp
index 21a707ba0..8825c6f24 100644
--- a/libc/bionic/pthread_create.cpp
+++ b/libc/bionic/pthread_create.cpp
@@ -65,8 +65,6 @@ bool __init_tls(pthread_internal_t* thread) {
     return false;
   }
 
-  prctl(PR_SET_VMA, PR_SET_VMA_ANON_NAME, allocation, allocation_size, "bionic TLS guard");
-
   // Carve out the writable TLS section.
   thread->bionic_tls = reinterpret_cast<bionic_tls*>(static_cast<char*>(allocation) +
                                                      PTHREAD_GUARD_SIZE);
@@ -77,7 +75,6 @@ bool __init_tls(pthread_internal_t* thread) {
     return false;
   }
 
-  prctl(PR_SET_VMA, PR_SET_VMA_ANON_NAME, thread->bionic_tls, BIONIC_TLS_SIZE, "bionic TLS");
   return true;
 }
 
@@ -105,7 +102,6 @@ void __init_alternate_signal_stack(pthread_internal_t* thread) {
     // We can only use const static allocated string for mapped region name, as Android kernel
     // uses the string pointer directly when dumping /proc/pid/maps.
     prctl(PR_SET_VMA, PR_SET_VMA_ANON_NAME, ss.ss_sp, ss.ss_size, "thread signal stack");
-    prctl(PR_SET_VMA, PR_SET_VMA_ANON_NAME, stack_base, PTHREAD_GUARD_SIZE, "thread signal stack guard");
   }
 }
 
@@ -184,8 +180,6 @@ static void* __create_thread_mapped_space(size_t mmap_size, size_t stack_guard_s
     munmap(space, mmap_size);
     return NULL;
   }
-  prctl(PR_SET_VMA, PR_SET_VMA_ANON_NAME, space, stack_guard_size, "thread stack guard");
-
   return space;
 }
 
-- 
2.17.1

